File: src/PointCoverage.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

class Segment implements Comparable<Segment> {
    private int left;
    private int right;

    int getRight() {
        return right;
    }

    void setLeft(int left) {
        this.left = left;
    }

    void setRight(int right) {
        this.right = right;
    }

    boolean contains(int num) {
        return num >= left && num <= right;
    }


    @Override
    public int compareTo(Segment s) {
        if (this.right == s.right) {
            return 0;
        }
        if (this.right > s.right) {
            return 1;
        }
        return -1;
    }
}

public class PointCoverage {
    public static void main(String[] args) {
        getPoints(getInput());
    }

    private static Segment[] getInput() {
        Scanner scanner = new Scanner(System.in);
        int numberOfSegments = scanner.nextInt();
        Segment[] segments = new Segment[numberOfSegments];
        for (int i = 0; i < numberOfSegments; i++) {
            segments[i] = new Segment();
            segments[i].setLeft(scanner.nextInt());
            segments[i].setRight(scanner.nextInt());
        }
        return segments;
    }

    private static void getPoints(Segment[] segments) {
        Arrays.sort(segments);
        ArrayList<Segment> segmentsList = new ArrayList<>(Arrays.asList(segments));
        ArrayList<String> result = new ArrayList<>();
        int count = 0;
        while (segmentsList.size() > 0) {
            int point = segmentsList.get(0).getRight();
            result.add(String.valueOf(point));
            count++;
//            segmentsList.remove(0);
            while (segmentsList.size() > 0 && segmentsList.get(0).contains(point)) {
                segmentsList.remove(0);
            }
        }
        System.out.println(count);
        System.out.println(String.join(" ", result));
    }
}

File: src/point_coverage.py
class Segment:
    def __init__(self, left, right):
        self.left = left
        self.right = right

    def __contains__(self, item):
        return self.left <= item <= self.right

    def __repr__(self):
        return "left: {}, right: {}".format(self.left, self.right)


def run():
    segments = [Segment(*map(int, input().split())) for _ in range(int(input()))]
    segments.sort(key=lambda s: s.right)
    result = []
    point = 0
    for segment in segments:
        if not result or (point not in segment):
            point = segment.right
            result.append(point)
    print(len(result))
    print(' '.join(map(str, result)))


if __name__ == '__main__':
    run()
